### PR TITLE
Add snap to closest point option for sample

### PR DESCRIPTION
**Pull request by MatthewFlamm**
### Overview
Adds new 9.3.0 feature to snap to cell with closest point when a cell isn't found.

### Details

vtk uses naming "SnapToCellWithClosestPoint", which is a very descriptive name, but I think this is too wordy for a python parameter with underscores.

### CODE REVIEW

**Review by tkoyama010 — DISMISSED** [review]
LGTM. Thanks!

**Review by banesullivan — APPROVED** [review]

**Comment by MatthewFlamm** [comment]
Thanks all.  I'd love to sneak this into the next release, but it isn't that important either way.

File: pyvista/core/filters/data_set.py [hunk 1]
```diff
@@ -3641,6 +3648,11 @@ def sample(
                     ) from err
             alg.SetCellLocatorPrototype(locator)
 
+        if snap_to_closest_point:
+            try:
+                alg.SnapToCellWithClosestPointOn()
+            except AttributeError:
```
Comment (tkoyama010) [inline pyvista/core/filters/data_set.py:3654]:
```suggestion
            except AttributeError:  # pragma: no cover
```

**Review by tkoyama010 — APPROVED** [review]
